platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/admin/core/api/ConfigurationAdmin.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.core.api;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.shiro.subject.Subject;
+import org.osgi.framework.Bundle;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.metatype.ObjectClassDefinition;
+
+/**
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public interface ConfigurationAdmin {
+
+    /**
+     * Returns the {@link Configuration} associated with the given PID
+     * @param pid - service pid for the configuration
+     * @return Configuration
+     */
+    Configuration getConfiguration(String pid);
+
+    /**
+     * Returns the {@link ObjectClassDefinition} associated with the given {@link Configuration}.
+     * The ObjectClassDefinition is the actual configuration part of the metatype that holds all
+     * of the configurable pieces.
+     * @param config - The configuration associated with the metatype
+     * @return ObjectClassDefinition
+     */
+    ObjectClassDefinition getObjectClassDefinition(Configuration config);
```

Comment:
I could do that, but since this is experimental anyways, I decided it wasn't worth it at this point. If there are enough dissenting opinions, I can wrap it with something.